solc/CommandLineInterface.cpp

```diff
@@ -2019,7 +1087,7 @@ void CommandLineInterface::outputCompilationResults()
 
 	if (
 		!m_compiler->compilationSuccessful() &&
-		m_stopAfter == CompilerStack::State::CompilationSuccessful
+		m_options.stopAfter == CompilerStack::State::CompilationSuccessful
 	)
```

Comment:
I'll try. But this is something I'll have to test manually because it's not covered by tests.

Reply:
The problem here is that's in `CommandLineInterface.cpp` and most of that file remains hard to test (except maybe some static functions) because it just prints stuff to stdout. It would have to be refactored further to make the public functions return data in some structured form which then becomes the input for the functions that print the output. Basically, just like I extracted the parser, we should also extract the result formatter. I actually thought about refactoring it that way but then this PR would be even bigger so I dropped the idea for now.